src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Android.cs

```diff
@@ -0,0 +1,409 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System
+{
+    public sealed partial class TimeZoneInfo
+    {
+        private const string TimeZoneFileName = "tzdata";
+
+        private static AndroidTzData tzData = new AndroidTzData();
+
+        // This should be called when name begins with GMT
+        private static int ParseGMTNumericZone(string name)
+        {
+            int sign;
+            if (name[3] == '+')
+            {
+                sign = 1;
+            }
+            else if (name[3] == '-')
+            {
+                sign = -1;
+            }
+            else
+            {
+                return 0;
+            }
+
+            int where;
+            int hour = 0;
+            bool colon = false;
+            for (where = 4; where < name.Length; where++)
+            {
+                char c = name[where];
+
+                if (c == ':')
+                {
+                    where++;
+                    colon = true;
+                    break;
+                }
+
+                if (c >= '0' && c <= '9')
+                {
+                    hour = hour * 10 + c - '0';
+                }
+                else
+                {
+                    return 0;
+                }
+            }
+
+            int min = 0;
+            for (; where < name.Length; where++)
+            {
+                char c = name [where];
+
+                if (c >= '0' && c <= '9')
+                {
+                    min = min * 10 + c - '0';
+                }
+                else
+                {
+                    return 0;
+                }
+            }
+
+            if (colon)
+            {
+                return sign * (hour * 60 + min) * 60;
+            }
+            else if (hour >= 100)
+            {
+                return sign * ((hour / 100) * 60 + (hour % 100)) * 60;
+            }
+            else
+            {
+                return sign * (hour * 60) * 60;
+            }
+        }
+
+        private static TimeZoneInfo? GetTimeZone(string id, string name)
+        {
+            if (name == "GMT" || name == "UTC")
```

Comment:
It might be faster to use `String.Compare (name, "GMT", StringComparison.Ordinal)` here, the equals operator may use the current culture for comparison.

Reply:
> the equals operator may use the current culture for comparison.

It does not and it's faster than String.Compare as it calls down to SequenceEquals.